src-tauri/src/data_conversion/distance_conversion.rs

```diff
@@ -12,10 +12,16 @@ use crate::mesh::device::MeshNode;
 * Altitude: in meters above sea level, no conversion needed
 */
 pub fn get_distance(node_1: MeshNode, node_2: MeshNode) -> f64 {
+    let default = 0;
     let node_1_data = node_1.data;
     let node_2_data = node_2.data;
-    let node_1_pos = node_1_data.position.unwrap();
-    let node_2_pos = node_2_data.position.unwrap();
+    let has_node_1_pos = node_1_data.position;
+    let has_node_2_pos = node_2_data.position;
+    if has_node_1_pos.is_none() || has_node_2_pos.is_none() {
+        return default as f64;
+    }
+    let node_1_pos = has_node_1_pos.unwrap();
```

Comment:
We shouldn't unwrap this, if there's a chance of failure we should either return an error or an option

Reply:
Switched to match statements to use data from the nested struct (unwrapping directly from references in the hashmap).

Docs are [here](https://doc.rust-lang.org/stable/book/ch18-03-pattern-syntax.html)